x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -38,9 +38,13 @@
 public class StackTemplateRegistry extends IndexTemplateRegistry {
     private static final Logger logger = LogManager.getLogger(StackTemplateRegistry.class);
 
-    // Current version of the registry requires all nodes to be at least 8.9.0.
+    // Historical node feature kept here as LegacyStackTemplateRegistry is deprecated
     public static final NodeFeature STACK_TEMPLATES_FEATURE = new NodeFeature("stack.templates_supported");
 
+    // this node feature is a redefinition of {@link DataStreamFeatures#SUPPORTS_DATA_STREAM_LIFECYCLE} and it's meant to avoid adding a
+    // dependency to the data-streams module just for this
+    public static final NodeFeature SUPPORTS_DATA_STREAM_LIFECYCLE = new NodeFeature("supports_data_stream_lifecycle");
```

Comment:
Same suggestion here
```suggestion
    public static final NodeFeature DATA_STREAM_LIFECYCLE = new NodeFeature("data_stream.rollover.lifecycle");
```